packages/vm/src/testing/mock.rs

```diff
@@ -182,9 +189,12 @@ impl<C: DeserializeOwned> MockQuerier<C> {
 impl<C: DeserializeOwned> Querier for MockQuerier<C> {
     fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
         let res = self.querier.raw_query(bin_request);
-        let used_gas = (bin_request.len() + to_binary(&res).unwrap().len()) as u64;
+        // FIXME: is this correct? Was this charged already?
```

Comment:
what do you mean by these fixmes? it seems like this is correct

Reply:
To be honest, I have no idea how gas is charged when the querier is used

Reply:
When we call out to the querier, all gas used in that operation is charged to the sdk gas meter (and not to this instance).

And we do properly pass this "externally used gas" back to the rust side: https://github.com/CosmWasm/go-cosmwasm/blob/master/api/callbacks.go#L414